### PR TITLE
build and release theia docker image

### PR DESCRIPTION
**Pull request by EvanBoyle**
This builds a [theia](https://theia-ide.org/) docker image every time we merge to master.

This allows us to have an image that we can link to in our gettings started documentation for users who would prefer to use docker and a local, web-based IDE, rather than installing the whole toolchain. 

Once we publish the image, you can run it via:

```sh
docker run -it --init -p 3000:3000 pulumi/theia:latest
```

`localhost:3000` will have a web-based IDE with terminal, full pulumi toolchain, and all of our examples. 

Screen Shot 2020-02-21 at 10 45 49 AM

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+FROM pulumi/pulumi	
+
```
Comment (lukehoban) [inline Dockerfile]: Perhaps move these theia files to `misc/theia`?

File: docker.package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{
+    "private": true,
+    "theia": {
+        "frontend": {
+            "config": {
+                "applicationName": "Theia JavaScript/TypeScript Example",
```
Comment (lukehoban) [inline docker.package.json]: Should this be different?

File: docker.package.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    },
+    "resolutions": {
+        "vscode-languageserver-protocol": "3.15.0-next.9",
+        "vscode-languageserver-types": "3.15.0-next.5",
+        "**/vscode-json-languageserver/**/vscode-languageserver": "6.0.0-next.1"
+    },
+    "devDependencies": {
+        "@theia/cli": "latest"
+    }
+}
```
Comment (lukehoban) [inline docker.package.json]: This and elsewhere -fix trailing new lines?

**Comment by stack72** [comment]
Hey @EvanBoyle 

As per cleanup, I am going to close this out - it's gone stale. If you want to re-address this then we can definitely do that

P.

**Pull request closed without merging**